app/assets/javascripts/exploreApis.js

```diff
@@ -51,65 +38,166 @@ export function onPageShow() {
+                setVisible(elApiResultsSize, visible);
+            },
+            set enableFilters(enabled) {
+                document.querySelectorAll('input[type=checkbox]').forEach(checkbox => checkbox.disabled = !enabled);
             }
         };
     })();
 
-    const paginator = buildPaginator(15)
+    function updateHiddenByPaginationValues(itemsVisibility) {
+        itemsVisibility.forEach(([el, visibleOnCurrentPage]) => {
+            const api = model.getApiForElement(el);
+            api.hiddenByPagination = !visibleOnCurrentPage;
+        });
+        view.setApiPanelVisibility(model.apis);
+    }
+
+    const paginator = buildPaginator(15, updateHiddenByPaginationValues)
+    const model = buildModel(view.apiDetailPanels);
 
     function buildFilterFunctions() {
         return filters.map(filter=> filter.buildFilterFunction());
     }
 
-    const model = {
-        apis: view.apiDetailPanels.map(el => ({
-            data: {
-                apiStatus: el.dataset['apistatus'],
-                domain: el.dataset['domain'],
-                subdomain: el.dataset['subdomain'],
-                hods: new Set(el.dataset['hods'].split(',').filter(h => h)),
-                apiName: el.dataset['apiname'],
-                platform: el.dataset['platform']
-            },
-            el,
-            hiddenByFilters: false
-        })),
-        get resultCount() {
-            return this.apis.filter(apiDetail => ! apiDetail.hiddenByFilters).length;
-        }
-    };
-
-    function applyFilters() {
+    function applyFiltersAndPagination() {
         const filterFns = buildFilterFunctions();
         model.apis.forEach(apiDetail => {
             apiDetail.hiddenByFilters = ! filterFns.every(fn => fn(apiDetail.data));
         });
 
-        view.setApiPanelVisibility(model.apis);
-        paginator.render(model.apis.filter(apiDetail => ! apiDetail.hiddenByFilters).map(panel => panel.el));
+        const apiPanelsToShowAfterFiltering = model.apis.filter(apiDetail => apiDetail.includeInResults).map(panel => panel.el);
+        paginator.render(apiPanelsToShowAfterFiltering);
 
         view.setResultCount(model.resultCount);
-        view.toggleNoResultsPanel(model.resultCount === 0);
+
+        const filteredCount = model.filteredCount;
+        if (model.currentSearchText) {
+            const showingAllResults = filteredCount === 0;
+            view.showSearchResultsPanelSuccess(showingAllResults, model.searchResultCount, model.currentSearchText);
+        } else {
+            view.hideSearchResultsPanel();
+        }
+
+        if (model.filteredCount > 0) {
+            view.showFilterResultsPanel(model.resultCount, model.filteredCount);
+        } else {
+            view.hideFilterResultsPanel();
+        }
     }
 
-    view.onFiltersChanged(() => {
-        applyFilters();
+    filters.forEach(filter => filter.initialise(model.apis));
+    filters.forEach(filter=> filter.onChange(applyFiltersAndPagination));
```

Comment:
not sure if this can be done this way:
```suggestion
    filters.forEach(filter => {
        filter.initialise(model.apis));
        filter.onChange(applyFiltersAndPagination);
    });
```